test/e2e/reboot.go

```diff
@@ -64,52 +64,60 @@ var _ = Describe("Reboot", func() {
 	})
 
 	It("should reboot each node and ensure they function upon restart", func() {
-		// This test requires SSH, so the provider check should be identical to
-		// there (the limiting factor is the implementation of util.go's
-		// getSigner(...)).
-		provider := testContext.Provider
-		if !providerIs("gce", "gke") {
-			By(fmt.Sprintf("Skipping reboot test, which is not implemented for %s", provider))
-			return
-		}
-
-		// Get all nodes, and kick off the test on each.
-		nodelist, err := c.Nodes().List(labels.Everything(), fields.Everything())
-		if err != nil {
-			Failf("Error getting nodes: %v", err)
-		}
-		result := make(chan bool, len(nodelist.Items))
-		for _, n := range nodelist.Items {
-			go rebootNode(c, provider, n.ObjectMeta.Name, result)
-		}
+		testReboot(c, true)
+	})
 
-		// Wait for all to finish and check the final result.
-		failed := false
-		// TODO(mbforbes): Change to `for range` syntax and remove logging once
-		// we support only Go >= 1.4.
-		for _, n := range nodelist.Items {
-			if !<-result {
-				Failf("Node %s failed reboot test.", n.ObjectMeta.Name)
-				failed = true
-			}
-		}
-		if failed {
-			Failf("Test failed; at least one node failed to reboot in the time given.")
-		}
+	It("should cause kernel panic on each node and ensure they function upon restart", func() {
+		testReboot(c, false)
 	})
 })
 
+func testReboot(c *client.Client, clean bool) {
```

Comment:
I've made reboot command a parameter, added the new cases suggested by you and also one more case (unclean reboot: echo b | sudo tee /proc/sysrq-trigger).

Regarding generalization of testReboot: I couldn't find anything like this in utils. I'm not sure if such functionality (with waiting for nodes to be ready/not ready) would not be too specific and therefore confusing. We can still consider introduce it in future in another PR.